Assets/MixedRealityToolkit.Tests/PlayModeTests/InputEventSystemTests.cs

```diff
@@ -0,0 +1,300 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+#if !WINDOWS_UWP
+// When the .NET scripting backend is enabled and C# projects are built
+// Unity doesn't include the required assemblies (i.e. the ones below).
+// Given that the .NET backend is deprecated by Unity at this point it's we have
+// to work around this on our end.
+using Microsoft.MixedReality.Toolkit.Diagnostics;
+using Microsoft.MixedReality.Toolkit.Input;
+using Microsoft.MixedReality.Toolkit.Utilities;
+using NUnit.Framework;
+using System.Collections;
+using System.Linq;
+using UnityEngine;
+using UnityEngine.TestTools;
+
+namespace Microsoft.MixedReality.Toolkit.Tests
+{
+    class InputEventSystemTests
+    {
+        // Can't use NUnit setup routine, because needs to skip several frames while initializing/removing objects.
+        public IEnumerator SetupMrtkWithoutCursors()
+        {
+            TestUtilities.InitializeMixedRealityToolkitAndCreateScenes(true);
+            TestUtilities.InitializePlayspace();
+
+            IMixedRealityInputSystem inputSystem = null;
+            MixedRealityServiceRegistry.TryGetService(out inputSystem);
+
+            Assert.IsNotNull(inputSystem, "Input system must be initialized");
+
+            // Let input system to register all cursors and managers.
+            yield return null;
+
+            // Switch off / Destroy all input components, which listen to global events
+            Object.Destroy(inputSystem.GazeProvider.GazeCursor as Behaviour);
+            inputSystem.GazeProvider.Enabled = false;
+
+            var diagnosticsVoiceControls = Object.FindObjectsOfType<DiagnosticsSystemVoiceControls>();
+            foreach (var diagnosticsComponent in diagnosticsVoiceControls)
+            {
+                diagnosticsComponent.enabled = false;
+            }
+
+            // Let objects be destroyed
+            yield return null;
+
+            // Check that input system is clean
+            Assert.AreEqual(((BaseEventSystem)inputSystem).EventHandlersByType.Count, 0, "Input event system handler registry is not empty in the beginning of the test.");
+
+            yield return null;
+        }
+
+        [TearDown]
+        public void ShutdownMrtk()
+        {
+            TestUtilities.ShutdownMixedRealityToolkit();
+        }
+
+        /// <summary>
+        /// </summary>
+        [UnityTest]
+        public IEnumerator Test01_TestObjectBasedEventRegistration()
```

Comment:
Nit: Is the Test01 prefix/Test02 prefix needed to ensure some sort of deterministic order? I left similar comments on other folks' reviews - these prefixes don't add a lot of value, and as we add/delete them they will immediately get out order/have numerical gaps.

Imagine we want to add another test that is related to object based event registration, but Test02 is already taken, in which case our naming also gets weird.

Reply:
Honestly I've added it after looking at other tests (edit mode, because there were very few play mode). I thought it was intentional. I'll remove them here.